### PR TITLE
Make security project ref

### PR DESCRIPTION
**Pull request by jkotalik**

### CODE REVIEW

File: src/Mvc/src/Microsoft.AspNetCore.Mvc.ViewFeatures/Microsoft.AspNetCore.Mvc.ViewFeatures.csproj [hunk 1]
```diff
@@ -13,13 +13,16 @@ Microsoft.AspNetCore.Mvc.ViewComponent</Description>
     <PackageTags>aspnetcore;aspnetcoremvc</PackageTags>
   </PropertyGroup>
 
+  <ItemGroup>
+    <Compile Include="$(RepositoryRoot)src\Shared\ChunkingCookieManager\**\*.cs" />
```
Comment (natemcmaster) [inline src/Mvc/src/Microsoft.AspNetCore.Mvc.ViewFeatures/Microsoft.AspNetCore.Mvc.ViewFeatures.csproj]:
```suggestion
    <Compile Include="$(MSBuildThisFileDirectory)..\..\..\Shared\ChunkingCookieManager\**\*.cs" />
```

Until MVC converts to the new ref style, RepositoryRoot will point to src/Mvc/

File: src/Mvc/test/Microsoft.AspNetCore.Mvc.FunctionalTests/Microsoft.AspNetCore.Mvc.FunctionalTests.csproj [hunk 2]
```diff
@@ -24,6 +24,10 @@
     <WebApplicationFactoryContentRootAttribute Include="FiltersWebSite" AssemblyName="FiltersWebSite" ContentRootPath="/../WebSites/FiltersWebSite" ContentRootTest="Incorrect.csproj" Priority="-1" />
   </ItemGroup>
 
+  <ItemGroup>
+    <Compile Include="$(RepositoryRoot)src\Shared\ChunkingCookieManager\**\*.cs" />
```
Comment (natemcmaster) [inline src/Mvc/test/Microsoft.AspNetCore.Mvc.FunctionalTests/Microsoft.AspNetCore.Mvc.FunctionalTests.csproj]: Same here.

File: src/Security/Authentication.Cookies/src/Microsoft.AspNetCore.Authentication.Cookies.csproj [hunk 3]
```diff
@@ -10,11 +10,11 @@
   </PropertyGroup>
 
   <ItemGroup>
-    <Compile Include="..\..\shared\Microsoft.AspNetCore.ChunkingCookieManager.Sources\**\*.cs" />
+    <Compile Include="$(RepositoryRoot)src\Shared\ChunkingCookieManager\**\*.cs" />
```
Comment (natemcmaster) [inline src/Security/Authentication.Cookies/src/Microsoft.AspNetCore.Authentication.Cookies.csproj]:
```suggestion
    <Compile Include="$(SharedSourceRoot)ChunkingCookieManager\**\*.cs" />
```

File: src/Security/Authentication.JwtBearer/src/Microsoft.AspNetCore.Authentication.JwtBearer.csproj [hunk 4]
```diff
@@ -9,11 +9,11 @@
   </PropertyGroup>
 
   <ItemGroup>
-    <ProjectReference Include="..\Microsoft.AspNetCore.Authentication\Microsoft.AspNetCore.Authentication.csproj" />
+    <Reference Include="Microsoft.AspNetCore.Authentication" />
   </ItemGroup>
```
Comment (natemcmaster) [inline src/Security/Authentication.JwtBearer/src/Microsoft.AspNetCore.Authentication.JwtBearer.csproj]: Nit: you can condense item groups of the same time.

File: src/Security/Authentication.WsFederation/samples/WsFedSample/WsFedSample.csproj [hunk 5]
```diff
@@ -0,0 +1,24 @@
+﻿<Project Sdk="Microsoft.NET.Sdk.Web">
+
+  <PropertyGroup>
+    <TargetFrameworks>net461;netcoreapp2.0</TargetFrameworks>
```
Comment (natemcmaster) [inline src/Security/Authentication.WsFederation/samples/WsFedSample/WsFedSample.csproj]:
```suggestion
    <TargetFrameworks>net461;netcoreapp2.1</TargetFrameworks>
```

Might have originally been 2.0, but let's cleanup as long as we're here :)

File: src/Security/Authentication/src/Microsoft.AspNetCore.Authentication.csproj [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+    <Reference Include="Microsoft.AspNetCore.DataProtection" />
+    <Reference Include="Microsoft.AspNetCore.Http" />
+    <Reference Include="Microsoft.AspNetCore.Http.Extensions" />
+    <Reference Include="Microsoft.Extensions.Logging.Abstractions" />
+    <Reference Include="Microsoft.Extensions.Options" />
+    <Reference Include="Microsoft.Extensions.WebEncoders" />
+  </ItemGroup>
+
+  <ItemGroup>
+    <PackageReference Include="Microsoft.Extensions.SecurityHelper.Sources" PrivateAssets="All" Version="$(MicrosoftExtensionsSecurityHelperSourcesPackageVersion)" />
```
Comment (natemcmaster) [inline src/Security/Authentication/src/Microsoft.AspNetCore.Authentication.csproj]:
```suggestion
    <Reference Include="Microsoft.Extensions.SecurityHelper.Sources" PrivateAssets="All" />
```

File: src/Security/Authorization.Policy/src/Microsoft.AspNetCore.Authorization.Policy.csproj [hunk 7]
```diff
@@ -9,11 +9,11 @@
...
 
   <ItemGroup>
-    <ProjectReference Include="..\Microsoft.AspNetCore.Authorization\Microsoft.AspNetCore.Authorization.csproj" />
+    <Reference Include="Microsoft.AspNetCore.Authorization" />
+    <Reference Include="Microsoft.AspNetCore.Authentication.Abstractions" />
   </ItemGroup>
 
   <ItemGroup>
-    <PackageReference Include="Microsoft.AspNetCore.Authentication.Abstractions" Version="$(MicrosoftAspNetCoreAuthenticationAbstractionsPackageVersion)" />
     <PackageReference Include="Microsoft.Extensions.SecurityHelper.Sources" PrivateAssets="All" Version="$(MicrosoftExtensionsSecurityHelperSourcesPackageVersion)" />
```
Comment (natemcmaster) [inline src/Security/Authorization.Policy/src/Microsoft.AspNetCore.Authorization.Policy.csproj]:
```suggestion
    <Reference Include="Microsoft.Extensions.SecurityHelper.Sources" PrivateAssets="All" />
```

File: src/Security/CookiePolicy/test/Microsoft.AspNetCore.CookiePolicy.Test.csproj [hunk 8]
```diff
@@ -0,0 +1,18 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <TargetFrameworks>$(StandardTestTfms)</TargetFrameworks>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <Compile Include="$(RepositoryRoot)src\Shared\ChunkingCookieManager\**\*.cs" />
```
Comment (natemcmaster) [inline src/Security/CookiePolicy/test/Microsoft.AspNetCore.CookiePolicy.Test.csproj]:
```suggestion
    <Compile Include="$(SharedSourceRoot)ChunkingCookieManager\**\*.cs" />
```

**Review by natemcmaster — CHANGES_REQUESTED** [review]
A good start. I added a few comments about little things that can change in the project file, and I pushed a commit to your PR to trim some stale content from the README in src/Security

One thing you should for sure revert is the removal of the git submodules. :) Still required for now.

cc @Tratcher - you may be interested in viewing the source code layout.

**Comment by Tratcher** [comment]
The layout's a little odd, especially for the AuthN samples and tests that pull in stuff from other folders. How about:
- Authentication/
  - src
    - Core
    - Cookies
    - Facebook
    - Google
    - JwtBearer
    - ...
  - samples/SocialSample 
  - tests
- Authorization/
  - src
    - Core (src, test)
    - Policy
- CookiePolicy
- Interop

Also, the src folder can be flattened when it's the only folder for a given package. E.g. `Authenticaiton/Facebook/Microsoft.AspnetCore.Authentication.Facebook.csproj`

**Comment by JunTaoLuo** [comment]
I'll take over this PR since I need it for Identity.

**Comment by Tratcher** [comment]
Updated after talking to @natemcmaster 

- Authentication/
  - Core
  - Cookies
  - Facebook
  - Google
  - JwtBearer
  - ...
  - samples/SocialSample 
  - tests
- Authorization/
  - Core
  - Policy
- CookiePolicy
- Interop

And don't worry about flattening the solo src dirs, it sounds like each of these will get paired with a ref dir.

**Comment by JunTaoLuo** [comment]
Updated

**Review by Tratcher — APPROVED** [review]

**Review by JunTaoLuo — APPROVED** [review]

**Review by natemcmaster — APPROVED** [review]

**Pull request merged**